Gem/Code/Include/NetworkPrefabSpawnerInterface.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Math/Transform.h>
+#include <AzFramework/Spawnable/SpawnableEntitiesInterface.h>
+
+namespace MultiplayerSample
+{
+    using PrefabSpawnCallbackBeforeActivation = AZStd::function<void(
+        AZStd::shared_ptr<AzFramework::EntitySpawnTicket>,
+        AzFramework::SpawnableEntityContainerView)>;
+
+    using PrefabSpawnCallback = AZStd::function<void(
+        AZStd::shared_ptr<AzFramework::EntitySpawnTicket>,
+        AzFramework::SpawnableConstEntityContainerView)>;
+
+    struct PrefabCallbacks
+    {
+        PrefabSpawnCallbackBeforeActivation m_beforeActivateCallback;
+        PrefabSpawnCallback m_onActivateCallback;
+    };
+
+    class NetworkPrefabSpawnerRequests
+    {
+    public:
+        AZ_RTTI(NetworkPrefabSpawnerRequests, "{82e5cfb5-6a1a-4bd1-b48d-cd817474d611}");
+        virtual ~NetworkPrefabSpawnerRequests() = default;
+
+        virtual void SpawnPrefab(const AZ::Transform& worldTm, const char* assetPath, PrefabCallbacks callbacks) = 0;
+        virtual void SpawnPrefabAsset(const AZ::Transform& worldTm, const AZ::Data::Asset<AzFramework::Spawnable>& asset, PrefabCallbacks callbacks) = 0;
+        virtual void SpawnDefaultPrefab(const AZ::Transform& worldTm, PrefabCallbacks callbacks) = 0;
+    };
+
+    class NetworkPrefabSpawnerTraits
+        : public AZ::ComponentBus
+    {
+    };
+
+    using NetworkPrefabSpawnerRequestBus = AZ::EBus<NetworkPrefabSpawnerRequests, NetworkPrefabSpawnerTraits>;
```

Comment:
This looks pretty clean aliasing in this manner, but it seems like we could really use some project wide conventions around all this for API consistency.

Reply:
I actually took from the gem templates you get when creating a new gem. So this looks like an implicit convention going forward now.